Task2_3_TrainingClass.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2_3
+{   
+    public class Training
+    {
+        private Lesson[] lessons;
+
+        public Training()
+        {
+            lessons = new Lesson[0];
+        }
+        public void Add(Lesson lesson)
+        {
+            Array.Resize(ref lessons, lessons.Length + 1);
+            lessons[lessons.Length - 1] = lesson;
+        }
+        public bool IsPractical()
+        {
+            foreach (var lesson in lessons)
+            {
+                if (lesson is Lecture)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+        public Training Clone()
+        {
```

Comment:
Implement ICloneable interface for Lessons, and code will be:
 foreach (var lesson in lessons)
 {
     clonedTraining.Add((ILesson)lesson.Clone());
 }

Reply:
corrected